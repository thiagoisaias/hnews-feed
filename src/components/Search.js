import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const FormContainer = styled.form`
  @media (max-width: 660px) {
    margin: 0 auto;
  }
`;

const FormInput = styled.input`
  height: 40px;
  font-size: 14px;
  line-height: 0;
  border: none;
  border-radius: 3px;
  outline: none;
  padding: 0 12px;
  margin-right: 8px;
  color: #484848;
  cursor: default;

  &::placeholder {
    color: #ccc;
  }

  @media (max-width: 425px) {
    width: 180px;
  }
  @media (min-width: 426px) {
    width: 300px;
  }
`;

const FormButton = styled.button`
  height: 40px;
  font-size: 14px;
  border: none;
  border-radius: 3px;
  color: #fff;
  background-color: #ff742b;
  padding: 12px;
  outline: none;
  &:hover {
    background-color: #fc9055;
    cursor: pointer;
  }
`;

const Search = props => {
  const { value, onChange, onSubmit } = props;
  return (
    <FormContainer onSubmit={onSubmit}>
      <FormInput
        type="text"
        placeholder="Search stories by title, url or author"
        value={value}
        onChange={onChange}
      />
      <FormButton type="submit"> Search </FormButton>
    </FormContainer>
  );
};

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Search;
