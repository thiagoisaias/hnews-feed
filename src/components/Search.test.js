import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Search from "./Search";

describe("Search", () => {
  it("renders properly", () => {
    const props = { value: "", onChange: () => {}, onSubmit: () => {} };
    const SearchTest = shallow(<Search {...props} />);
    expect(shallowToJson(SearchTest)).toMatchSnapshot();
  });
});
