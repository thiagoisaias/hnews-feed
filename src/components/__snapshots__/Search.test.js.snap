// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    onChange={[Function]}
    placeholder="Search stories by title, url or author"
    type="text"
    value=""
  />
  <styled.button
    type="submit"
  >
     Search 
  </styled.button>
</styled.form>
`;
